tests/js/keyboardInputTests.js

```diff
@@ -0,0 +1,268 @@
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.tests.firstDiscovery.usKeymap");
+    fluid.registerNamespace("gpii.tests.firstDiscovery.keyboardInput");
+
+    gpii.tests.firstDiscovery.charFromKeypressTestCases = [
+        { keyCode: undefined, expected: ""},
+        { keyCode: null, expected: "" },
+        { keyCode: 33, expected: "!" },
+        { keyCode: 48, expected: "0" },
+        { keyCode: 65, expected: "A" },
+        { keyCode: 97, expected: "a" }
+    ];
+
+    jqUnit.test("charFromKeypress", function () {
+        fluid.each(gpii.tests.firstDiscovery.charFromKeypressTestCases, function (testcase) {
+            var msg = "keyCode " + testcase.keyCode + " expect \"" + testcase.expected + "\"";
+            var actual = gpii.firstDiscovery.charFromKeypress({ which: testcase.keyCode });
+            jqUnit.assertEquals(msg, testcase.expected, actual);
+        });
+    });
+
+    jqUnit.test("usKeymap.isShiftEvent", function () {
+        var keymap = gpii.firstDiscovery.usKeymap();
+        jqUnit.assertTrue("true for shift", keymap.isShiftEvent({ which: keymap.shiftKeyCode }));
+        jqUnit.assertFalse("false for non-shift", keymap.isShiftEvent({ which: 0 }));
+    });
+
+    gpii.tests.firstDiscovery.usKeymap.nonLowerCase = [
+        String.fromCharCode(96),
+        String.fromCharCode(123),
+        "A"
+    ];
+
+    jqUnit.test("usKeymap.isLowerCaseLetter", function () {
+        jqUnit.expect(26 + 3);
+        var keymap = gpii.firstDiscovery.usKeymap();
+        for (var code = 97; code <= 122; code++) {
+            var ch = String.fromCharCode(code);
+            jqUnit.assertTrue(ch, keymap.isLowerCaseLetter(ch));
+        }
+        fluid.each(gpii.tests.firstDiscovery.usKeymap.nonLowerCase, function (ch) {
+            jqUnit.assertFalse(ch, keymap.isLowerCaseLetter(ch));
+        });
+    });
+
+    gpii.tests.firstDiscovery.usKeymap.shiftTestCases = [
+        { ch: "0", canShift: true, shifted: ")" },
+        { ch: "/", canShift: true, shifted: "?" },
+        { ch: "A", canShift: false },
+        { ch: "?", canShift: false}
+    ];
+
+    jqUnit.test("usKeymap canShiftChar and getShiftedChar", function () {
+        jqUnit.expect((26 * 2) + 4 + 2);
+        var keymap = gpii.firstDiscovery.usKeymap();
+        for (var code = 97; code <= 122; code++) {
+            var ch = String.fromCharCode(code);
+            jqUnit.assertTrue(ch, keymap.canShiftChar(ch));
+            jqUnit.assertEquals(ch, ch.toUpperCase(), keymap.getShiftedChar(ch));
+        }
+        fluid.each(gpii.tests.firstDiscovery.usKeymap.shiftTestCases, function (testcase) {
+            jqUnit.assertEquals(testcase.ch, testcase.canShift, keymap.canShiftChar(testcase.ch));
+            if (testcase.canShift) {
+                jqUnit.assertEquals(testcase.ch, testcase.shifted, keymap.getShiftedChar(testcase.ch));
```

Comment:
It might be easier to understand the test if the message is expanded. Right now it just has the input char.